sdk/storage/Azure.Storage.Files/tests/FileClientTests.cs

```diff
@@ -57,34 +57,39 @@ public void FilePathsParsing()
             var builder1 = new FileUriBuilder(uri1);
             var fileClient1 = new FileClient(uri1);
             TestHelper.AssertCacheableProperty("file.txt", () => fileClient1.Name);
+            TestHelper.AssertCacheableProperty("dir1/dir2/file.txt", () => fileClient1.Path);
             Assert.AreEqual("file.txt", builder1.LastDirectoryOrFileName);
 
             // one directory
             Uri uri2 = new Uri("http://dummyaccount.file.core.windows.net/share/dir1/file.txt");
             var builder2 = new FileUriBuilder(uri2);
             var fileClient2 = new FileClient(uri2);
             TestHelper.AssertCacheableProperty("file.txt", () => fileClient2.Name);
+            TestHelper.AssertCacheableProperty("dir1/file.txt", () => fileClient2.Path);
             Assert.AreEqual("file.txt", builder2.LastDirectoryOrFileName);
 
             // trailing slash
             Uri uri3 = new Uri("http://dummyaccount.file.core.windows.net/share/dir1/file.txt/");
             var builder3 = new FileUriBuilder(uri3);
             var fileClient3 = new FileClient(uri3);
             TestHelper.AssertCacheableProperty("file.txt", () => fileClient3.Name);
+            TestHelper.AssertCacheableProperty("dir1/file.txt/", () => fileClient3.Path);
```

Comment:
This feels a little odd - why do we kill it from the name but not the path?

Reply:
I don't feel too strongly about this, but having a slash in the Name clearly doesn't make sense, whereas in a path it seems reasonable.

Reply:
Also, it is included in the DirectoryOrFilePath property on the builder.

Reply:
I think we should probably remove trailing slashes from DirectoryOrFilePath in FileUriBuilder.  I wouldn't expect a file path to end in a slash.

Reply:
Ok fixed.